Xamarin.Forms.Material.iOS/MaterialActivityIndicatorRenderer.cs

```diff
@@ -80,36 +98,17 @@ protected override void OnElementPropertyChanged(object sender, PropertyChangedE
 
 		protected override void SetBackgroundColor(Color color)
 		{
-			// Do not call base to avoid the actual background view changing color.
-			//base.SetBackgroundColor(color);
-
-			if (Control == null)
+			if (Control == null || _backgroundLayer == null)
 				return;
 
-			// TODO: Investigate whether or not we want to look for the track
-			//       layer and change the color. This will be brittle.
-			//       For now, just show/hide the track.
-
-			if (color.IsDefault)
-				Control.TrackEnabled = false;
-			else
-				Control.TrackEnabled = true;
-
-			// handle the actual background color next to the main color
-			UpdateColor();
+			_backgroundLayer.Path = UIBezierPath.FromArc(Control.Center, Control.Radius - Control.StrokeWidth / 2, 0, 360, true).CGPath;
```

Comment:
Is it possible to set this path only when the radius/stroke width changes? We are creating a new native object and a managed object here. Also, is it possible to have this layer "scale"? Not sure of iOS's powers, but can we create a circle layer and let iOS resize it? I don't think the actual indictor has properties for the radius and stroke width, so we might just have to update when the size changes.

Reply:
The code is based on the material ActivityIndicator component
https://github.com/material-components/material-components-ios/blob/stable/components/ActivityIndicator/src/MDCActivityIndicator.m#L570
in `MDCActivityIndicator` this path recreates every time whenever you change any properties. Therefore, when drawing layers, we also recalculate this path. This allows to correctly display it when you change sizes or colors.